### PR TITLE
General streamlining

### PR DESCRIPTION
**Pull request by RochelleKnight**
- Create end date table that can be used for cox models, venn diagrams and table 2
- Update venn diagrams to censor for follow up start and end dates
- Run table 1 for each outcome group so that the correct covariates are used
- Re-structure table 2 script to use follow up end date table & reduce number of actions so that each cohort can be run with one action

I have checked all the event counts and person days with the cox models and all numbers match.

### CODE REVIEW

File: analysis/descriptives/venn_diagram.R [hunk 1]
```diff
@@ -13,12 +13,15 @@
...
+library(data.table)
+library(readr)
+library(dplyr)
 
 args <- commandArgs(trailingOnly=TRUE)
 
 if(length(args)==0){
   # use for interactive testing
-  population <- "electively_unvaccinated"
+  population <- "vaccinated"
```
Comment (GenevieveCezard) [inline analysis/descriptives/venn_diagram.R:24]: Use of `population` here, see previous comment.

**Review by GenevieveCezard — COMMENTED** [review]
See a couple of very minor comments!

File: analysis/preprocess/create_follow_up_end_date.R [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+#These are the study start and end dates for the Delta era
+cohort_start_date <- as.Date("2021-06-01")
+cohort_end_date <- as.Date("2021-12-14")
+
+## Read in active analyses table and filter to relevant outcomes
+
+active_analyses <- read_rds("lib/active_analyses.rds")
+active_analyses <- active_analyses %>% filter(active == "TRUE") %>% select(outcome_variable)
+
+follow_up_end_dates <- function(cohort){
```
[the remark this answers is not in the document]
Comment (RochelleKnight) [reply analysis/preprocess/create_follow_up_end_date.R:29]: I have updated venn diagrams, follow up end dates table, table 2 and pre-process to use cohort_name for consistency. Table 1 uses cohort_name to specify the cohort and then population to specify whether its the whole population/covid population

File: analysis/descriptives/table_2.R [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+fs::dir_create(here::here("output", "review", "descriptives"))
+
+#delta period
+cohort_start = as.Date("2021-06-01", format="%Y-%m-%d")
+cohort_end = as.Date("2021-12-14", format="%Y-%m-%d")
+
+agebreaks <- c(0, 40, 60, 80, 111)
+agelabels <- c("18_39", "40_59", "60_79", "80_110")
+
+table_2_subgroups_output <- function(population){
```
[the remark this answers is not in the document]
Comment (RochelleKnight) [reply analysis/descriptives/table_2.R:44]: This has been updated to use cohort_name for consistency

File: analysis/descriptives/table_2.R [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+    analyses_of_interest <- rbind(analyses_of_interest,analyses_to_run)
+    
+  }
+  
+  analyses_of_interest$strata[analyses_of_interest$strata=="South_Asian"]<- "South Asian"
+  analyses_of_interest <- analyses_of_interest %>% filter(cohort_to_run == population)
+  
+
+  unexposed_person_days <- unexposed_event_count <- rep("NA", nrow(analyses_of_interest))
+  total_person_days <- post_exposure_event_count <- overall_ir <- overall_ir_lower <- overall_ir_upper <- rep("NA", nrow(analyses_of_interest))
```
[the remark this answers is not in the document]
Comment (RochelleKnight) [reply analysis/descriptives/table_2.R:129]: Yes, I've just updated the name to make it slightly clearer

File: analysis/descriptives/venn_diagram.R (hunk 1 above)
Comment (RochelleKnight) [reply analysis/descriptives/venn_diagram.R:24]: This has been updated as per your suggestion

**Comment by RochelleKnight** [comment]
@GenevieveCezard Thanks for your comments, I have updated as per your suggestions

File: analysis/preprocess/create_follow_up_end_date.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (GenevieveCezard) [reply analysis/preprocess/create_follow_up_end_date.R]: Thank you for these changes

File: analysis/descriptives/table_2.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (GenevieveCezard) [reply analysis/descriptives/table_2.R]: Thank you!

File: analysis/descriptives/venn_diagram.R (hunk 1 above)
Comment (GenevieveCezard) [reply analysis/descriptives/venn_diagram.R]: Thank you!

**Review by GenevieveCezard — APPROVED** [review]
Changes were added as suggested.

**Pull request merged**